graylog2-server/src/main/java/org/graylog/plugins/views/startpage/lastOpened/LastOpenedService.java

```diff
@@ -16,34 +16,40 @@
  */
 package org.graylog.plugins.views.startpage.lastOpened;
 
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBObject;
 import com.mongodb.DuplicateKeyException;
 import org.bson.types.ObjectId;
 import org.graylog.grn.GRNTypes;
 import org.graylog.plugins.views.search.permissions.SearchUser;
+import org.graylog.plugins.views.startpage.recentActivities.ActivityType;
+import org.graylog.plugins.views.startpage.recentActivities.RecentActivityEvent;
 import org.graylog.security.entities.EntityOwnershipService;
 import org.graylog2.bindings.providers.MongoJackObjectMapperProvider;
 import org.graylog2.database.MongoConnection;
 import org.graylog2.database.PaginatedDbService;
+import org.graylog2.users.events.UserDeletedEvent;
 import org.mongojack.DBQuery;
 import org.mongojack.WriteResult;
 
 import javax.inject.Inject;
 import java.util.Optional;
 
-/*
- * TODO: remove entity, if a user is deleted?
- */
 public class LastOpenedService extends PaginatedDbService<LastOpenedForUserDTO> {
```

Comment:
My comments from FavortiesService may apply to this service as well.

Reply:
added same modifications on both classes